tests/AppBundle/EventListener/CommentNotificationListenerTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tests\AppBundle\EventListener;
+
+use AppBundle\Entity\Comment;
+use AppBundle\Entity\Post;
+use AppBundle\Entity\User;
+use AppBundle\EventListener\CommentNotificationListener;
+use Prophecy\Argument;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Unit test for the event listener.
+ *
+ * See http://symfony.com/doc/current/book/testing.html#unit-tests
+ *
+ * Execute the application tests using this command (requires PHPUnit to be installed):
+ *
+ *     $ cd your-symfony-project/
+ *     $ ./vendor/bin/phpunit
+ *
+ * @author Michael COULLERET <michael.coulleret@gmail.com>
+ */
+class CommentNotificationListenerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testListenerOnCommentCreated()
```

Comment:
I will add the test on the subscribed.

I do not think it is very relevant to break into 3 pieces because its role and corresponds to a single feature: send an email with a link and a translation.

I think what you're asking will complicate for too much. [Let's be KISS.](https://en.wikipedia.org/wiki/KISS_principle)